#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=168:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=38gb

#PBS -N bbblast-markah
#PBS -q bigiron
##PBS -t 1-9
umask 007
set -eux

#file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/bioservices/p2ep/brassica/map/chromosome/file.txt )
#home=/gpfs/fs3/home/rreid2


echo "Launching Blastn "
module load ncbi-blast+

#cd $home

cd /lustre/groups/bioservices/p2ep/blueberry/markers/rep/

#makeblastdb -in ../genomic/newblerAssembly.fna -dbtype nucl
makeblastdb -in  ~/repeatmasker/RepBase19.06.fasta/allRepeats.fna -dbtype nucl
echo `pwd`
#echo ${file}

#/lustre/home/rreid2/sw/seqtk-master/seqtk seq -L 30  -a ../flashmerged/${file}.fastq > ${file}.fasta
#/lustre/sw/ncbi-blast+/2.2.25/bin/blastx -db ~/db/nr/nr.faa -query ${file}.fasta  -out ./blastx-${file}vsnr.out -num_threads 12  -evalue 1e-10 -soft_masking true  -outfmt 0

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/bioservices/p2ep/dhmri/run2-oat-brass-oct25-2013/brassica/130924HiSeq_Run_Sample_V_158_NoIndex_L002_R2_001-clip.fasta -query ../chromosome/${file}.fna  -out ./blastn-${file}vsv158-r2.out -num_threads 12  -evalue 1 -soft_masking false  -outfmt 6





#All markers
#blastn -db ../../genomic/asurca-nextera454.fna  -query ../blueberryMarkers.fna  -out ./blastn-markerVsnextera454.out -num_threads 12  -evalue 1e-25 -soft_masking true  -outfmt 6 -word_size 78


#only markers that were used, aka 394(?0
blastn -db  ~/repeatmasker/RepBase19.06.fasta/allRepeats.fna  -query /lustre/groups/p2ep/blueberry/markers/mahkah394.txt-mahkas.fna  -out ./blastn-markahVsRep19.out -num_threads 12  -evalue 1e-10  -outfmt 6 


echo "Fin"
