#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=368:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=88gb

#PBS -N bbblastBroccoli

##PBS -t 1-306

#ssid=$(sed -n -e "${PBS_ARRAYID}p" ~/old/blueberry/reads/hiseq/split/files.txt)


# launching compare-to-gff.prl on a single node so it can run till the cows come home
echo "Launching BlastP "

#cd $home
#module load ncbi-blast+

cd /lustre/groups/bioservices/p2ep/brassica/orthocluster/splitty

#makeDB

#/lustre/sw/ncbi-blast+/2.2.25/bin/makeblastdb -in all-rnaseq-modelgenes-refgenes.faa -dbtype prot



#/lustre/sw/ncbi-blast+/2.2.25/bin/makeblastdb -in 2ndquarter.faa -dbtype prot
#/lustre/sw/ncbi-blast+/2.2.25/bin/makeblastdb -in 3rdquarter.faa -dbtype prot
#/lustre/sw/ncbi-blast+/2.2.25/bin/makeblastdb -in 4rthquarter.faa -dbtype prot
#/lustre/sw/ncbi-blast+/2.2.25/bin/makeblastdb -in topquarter.faa -dbtype prot

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastp -db all.fasta -query all.fasta  -out out.blastp.format0.txt -num_threads 24  -evalue 1e-5 -soft_masking false  -outfmt 0

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastp -db all.fasta -query all.fasta  -out blastp-e25.format0.out -num_threads 24  -evalue 1e-25 -soft_masking false  -outfmt 0

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastp -db all-rnaseq-modelgenes-refgenes.faa -query all-rnaseq-modelgenes-refgenes.faa  -out blastp-e40.format0.out -num_threads 24  -evalue 1e-40 -soft_masking false  -outfmt 0

/lustre/sw/ncbi-blast+/2.2.25/bin/blastp -db 4rthquarter.faa -query 4rthquarter.faa  -out blastp-e40.4rth-4rth.out -num_threads 24  -evalue 1e-40 -soft_masking false  -outfmt 0
