#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=68:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=85500mb

#PBS -N aaalignreads-trinityStyle
##PBS -q bigiron


#Number of times to run this script
#PBS -t 1-21

set eu
umask 007

##home=/gpfs/fs3/home/rreid2
#cd $HOME/blueberry/reads/illumina/091230Sequence1/
#export PATH=/sw/bowtie/bin/:$HOME/bin/sopra/source_codes_v1.4.3/SOPRA_with_prebuilt_contigs:$PATH
#file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/oat/badassay/fastq/file.txt)
file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/blueberry/rnaseq/sra-srr118/gage/file.txt)


module load blat

cd /group/p2ep/blueberry/rnaseq/sra-srr118/blat2genome/

#create index of reference
#bwa index -p ${file}.idx -a bwtsw /group/p2ep/blueberry/rnaseq/sra-srr118/cdhit/${file}-cdhit

module load htslib/1.1-1
module load samtools
################   Success was part below #####



#### To make .pslx file
blat -minIdentity=92  -tileSize=12 -extendThroughN -t=dna -q=dna -out=pslx -maxIntron=5000 /group/p2ep/blueberry/rnaseq/bigbamtheory/V_corymbosum_Aug_2015.fa /group/p2ep/blueberry/rnaseq/sra-srr118/finalTri-${file}.fna blat-${file}.pslx
~/scripts/perl/blat2gff.pl < blat-${file}.pslx > blat-${file}.gff


#######  To make PSL file and gff
blat -minIdentity=92  -tileSize=12 -extendThroughN -t=dna -q=dna -maxIntron=5000 /group/p2ep/blueberry/rnaseq/bigbamtheory/V_corymbosum_Aug_2015.fa /group/p2ep/blueberry/rnaseq/sra-srr118/finalTri-${file}.fna blat-${file}.psl
~/scripts/perl/blat2gff.pl < blat-${file}.psl > blat-${file}.gff
