#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=68:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=85500mb

#PBS -N hhhisat-bigbam
##PBS -q bigiron


#Number of times to run this script
##PBS -t 1-25
##PBS -t 10

set eu
umask 007

##home=/gpfs/fs3/home/rreid2
#cd $HOME/blueberry/reads/illumina/091230Sequence1/
#export PATH=/sw/bowtie/bin/:$HOME/bin/sopra/source_codes_v1.4.3/SOPRA_with_prebuilt_contigs:$PATH
file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/blueberry/rnaseq/sra-srr118/fileHiseq.txt)

module load bwa
module load htslib/1.1-1 samtools
cd /group/p2ep/blueberry/rnaseq/bigbamtheory 

#create index of reference

#module load samtools
#file=bwa-bb-bac15
################   Success was part below #####
#################################################		Removed the -I from bwa aln and this solves the alingment issue (casava 1.3+  vs 1.5+   #############
#bwa index -p GARM ../../genomic/garmAssembly.fna
#bwa aln  -t 12 GARM ../bb-011215/all6.fna > ${file}.aln.sai
#bwa mem GARM ${file}.aln.sai ../bb-011215/all6.fna  >  ${file}.sam

#module load bowtie2

#bowtie2-build  /lustre/groups/p2ep/blueberry/genomic/masurca-nextera454.fna ./nextera-bowtie2
#bowtie2  -f -x nextera-bowtie2 ../bb-011215/all6.fna -S ${file}.sam
#bowtie2 -S -q -p 12 ./allreads /lustre/groups/p2ep/GBS_data/barcodeSplit/Fla4BxW85/good/${file}.fastq ${file}.sam

RM=/home/rreid2/sw/hisat-0.1.4-beta
export PATH="${RM}:${PATH}"



#hisat-build /lustre/groups/p2ep/blueberry/genomic/sspace-take6.fna ./take6Index
#hisat  -p 24  -x ./truncTake6 -U ./allSRA.fastq -S bigbam.sam


 
samtools view -bS bigbam.sam | samtools sort - bigbam-sort



 
