#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=168:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
##PBS -l pmem=7600mb
##PBS -l vmem=400gb
#PBS -l vmem=90gb 

#PBS -N triiiiOatParents
##PBS -q bigiron

#Number of times to run this script
##PBS -t 1-3
##PBS -t 2

#file=$(sed -n -e "${PBS_ARRAYID}p" ~/echinoderms/dl/120727B2/files.txt)
#file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/rnaseq/parents/trinity/file.txt)

set eu
umask 007

module load bowtie
#module load oases
dir=tryBAM
file=BAM004

cd /lustre/groups/p2ep/rnaseq/parents/trinity/tryBAM


echo `pwd`

#perl ~/sw/trinityrnaseq_r2013-02-25/Trinity.pl --seqType fq --JM 400G --output ${file} --single /lustre/groups/p2ep/rnaseq/parents/fastx/${file}-filteredCombined.fastq --CPU 12  --full_cleanup
######  Above was a great success   ######

############   GET READ COUNTS , AKA GET ABUNDANCE     ##################
perl  ~/sw/trinityrnaseq_r2013-02-25/util/RSEM_util/run_RSEM_align_n_estimate.pl --transcripts ../${file}.Trinity.fasta  --seqType fq --single ../../fastx/${file}-filteredCombined.fastq --thread_count 12 --prefix ${file} 
echo "Fin"
