#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=68:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=85500mb

#PBS -N bbb-bwaTest 
##PBS -q bigiron


#Number of times to run this script
#PBS -t 1-44

set eu
umask 007

##home=/gpfs/fs3/home/rreid2
#cd $HOME/blueberry/reads/illumina/091230Sequence1/
#export PATH=/sw/bowtie/bin/:$HOME/bin/sopra/source_codes_v1.4.3/SOPRA_with_prebuilt_contigs:$PATH
file=$(sed -n -e "${PBS_ARRAYID}p" ~/cheatam/file.txt)

module load bwa

cd ~/cheatam/bwa/

#create index of reference
#bwa index -p NEXTERA454.idx -a bwtsw /lustre/groups/p2ep/blueberry/genomic/masurca-nextera454.fna

module load htslib/1.1-1
module load samtools
#file=bwa-bb-mahkah
################   Success was part below #####
#################################################		Removed the -I from bwa aln and this solves the alingment issue (casava 1.3+  vs 1.5+   #############
#bwa aln  -t 12 ~/db/human/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa ../UNC_Cheatham_121012_Run_${file}_R1_001.fastq > Sample_10_2079.aln.sai
bwa mem -t 12  ~/db/human/Homo_sapiens/UCSC/hg19/Sequence/BWAIndex/genome.fa  ../UNC_Cheatham_121012_Run_${file}_R1_001.fastq ../UNC_Cheatham_121012_Run_${file}_R2_001.fastq  >  ${file}.sam

#samtools import /lustre/groups/p2ep/blueberry/genomic/newblerAssembly.fna Fla4BxW85-385vsnewbler.sam  Fla4BxW85-385vsnewbler.bam

#samtools view -F 0x04 -b Fla4BxW85-385vsnewbler.bam > Fla4BxW85-385vsnewbler.rmd.bam
#samtools sort Fla4BxW85-385vsnewbler.rmd.bam  Fla4BxW85-385vsnewbler.rmd.bam.srt
samtools view -bS ${file}.sam | samtools sort - ${file}-sort

#samtools index ${file}-sort.bam

#samtools mpileup -ugf /lustre/groups/p2ep/oat/badassay/WG00014_OAT.targetSeq.fa ${file}-sort.bam | bcftools view -bvcg - > ${file}.bcf
#bcftools view ${file}.bcf > ${file}.vcf 

 
