#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=168:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=1

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=8gb

#PBS -N mmmergeReads-cote

##PBS -t 2-18
umask 007
set -eu

#file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/bioservices/gregory/HSMS_104_Cote/tophat/file.txt)

#echo "Launching tophat on ${file}"

#cd $home
#module load tophat
#module load bowtie2
#module load cufflinks
export PATH=$HOME/sw/HTSeq-0.6.1p1/scripts:$PATH
export PATH=/lustre/home/rreid2/anaconda/bin:$PATH


cd /lustre/groups/bioservices/gregory/HSMS_104_Cote/readcounts/

which htseq-count
which python

#htseq-count --stranded=no -f bam ../sam${file}/accepted_hits.bam /lustre/home/rreid2/db/human/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf > readcount-${file}.txt
for i in 8 1 14 2 20 23 15 19 5 4 24 9 16 21 6 3 10
do
	echo $i
	perl ~/scripts/perl/mergeHTSEQreads2table.pl readcountMasterTable.txt ../tophat/cufflinks_sam${i}/readcount-${i}.txt ${i}
	cp readcountMasterTable.txt.new readcountMasterTable.txt
done






echo "fin"
