#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=96:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=9

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l mem=88gb
#PBS -j oe
#PBS -N bbbeeeWA-tomato
#PBS -q steelhead
#PBS -t 1-12
umask 007
set -eu

file=$(sed -n -e "${PBS_ARRAYID}p" /nobackup/banana_genome/tomato/sra/srafiles.txt)
#dir=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/lorainelab/data/illumina/sweet_potato/filtered/dir.txt)

echo "Launching BWA alignment"

#module load java8 
module load samtools
module load bwa
module load bcftools


cd /nobackup/banana_genome/tomato/sra/ 

#file=nagcarlang
#bwa index -p TOMATO3 -a bwtsw /nobackup/banana_genome/tomato/publicData/cornell-wholegenome/S_lycopersicum_chromosomes.3.00.fa 
bwa mem -t 9  -k 11 -B 2 -O 5 -U 5   \
	../align/TOMATO3 \
	${file}.fasta  > ${file}.sam
samtools view -bS ${file}.sam | samtools sort -T ${file} -o ${file}-sorted.bam
samtools index ${file}-sorted.bam

#### MAKE VCF     ######
samtools mpileup  --count-orphans --skip-indels --output ${file}.bcf  \
        -uf /nobackup/banana_genome/tomato/publicData/cornell-wholegenome/S_lycopersicum_chromosomes.3.00.fa ${file}-sorted.bam
bcftools view ${file}.bcf > ${file}.vcf

########    READGROUPS    do this during BWA mem   -R     Foo fo foo fooof   
#R STR	Complete read group header line. ’\t’ can be used in STR and will be converted to a TAB in the output SAM. The read group ID will be attached to every read in the output. An example is ’@RG\tID:foo\tSM:bar’. [null]






















#bwa aln  -t 12 PAHANG.idx /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R1_paired.fq > ${file}-1.aln.sai
#bwa aln  -t 12 PAHANG.idx /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R2_paired.fq > ${file}-2.aln.sai
#bwa sampe PAHANG.idx ${file}-1.aln.sai ${file}-2.aln.sai -P \
# /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R1_paired.fq \
# /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R2_paired.fq  >  ${file}.sam

#samtools view -bS ${file}.sam | samtools sort -T ${file} -o ${file}-sorted.bam
#samtools mpileup -ugf /scratch/rreid2/banana/db/acuminata/GCA_000313855.2_ASM31385v2_genomic.fna ${file}-sorted.bam | bcftools view -vcg - > ${file}.bcf
#bcftools view ${file}.bcf > ${file}.vcf


#java -jar ~/sw/trimmomatic/Trimmomatic-0.36/trimmomatic-0.36.jar PE -threads 24 ../../trimgalore/${file}_R1_001_val_1.fq ../../trimgalore/${file}_R2_001_val_1.fq output_${file}-R1_paired.fq output_${file}-R1_unpaired.fq output_${file}-R2_paired.fq output_${file}-R2_unpaired.fq ILLUMINACLIP:TruSeq2-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50








