#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=20:30:23

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=85500mb

#PBS -N aaalignreads-trinity2.2
#PBS -q bigiron


#Number of times to run this script
#PBS -t 1-21

set eu
umask 007

##home=/gpfs/fs3/home/rreid2
#cd $HOME/blueberry/reads/illumina/091230Sequence1/
#export PATH=/sw/bowtie/bin/:$HOME/bin/sopra/source_codes_v1.4.3/SOPRA_with_prebuilt_contigs:$PATH
#file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/oat/badassay/fastq/file.txt)
file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/blueberry/rnaseq/sra-srr118/gage/file.txt)


module load bwa
module load bowtie2
module load bowtie
source ~/.bashrc

cd /lustre/groups/p2ep/blueberry/rnaseq/sra-srr118/gage/trinity2.2

#create index of reference
#bwa index -p ${file}.idx -a bwtsw /group/p2ep/blueberry/rnaseq/sra-srr118/cdhit/${file}-cdhit

module load htslib/1.1-1
module load samtools
################   Success was part below #####
#################################################		Removed the -I from bwa aln and this solves the alingment issue (casava 1.3+  vs 1.5+   #############


~/sw/trinity/trinityrnaseq-2.2.0/util/align_and_estimate_abundance.pl \
--transcripts /group/p2ep/blueberry/rnaseq/sra-srr118/cdhit/${file}-cdhit --seqType fa \
--single  /group/p2ep/blueberry/rnaseq/sra-srr118/filteredFasta/${file}_fastx-trim.fna \
--thread_count 12 \
--est_method RSEM --aln_method bowtie2 --trinity_mode --prep_reference --output_dir ${file}


#samtools import /lustre/groups/p2ep/blueberry/genomic/newblerAssembly.fna Fla4BxW85-385vsnewbler.sam  Fla4BxW85-385vsnewbler.bam

#samtools view -bS ${file}.sam | samtools sort - ${file}-sort

#samtools index ${file}-sort.bam

#samtools mpileup -ugf /lustre/groups/p2ep/oat/badassay/WG00014_OAT.targetSeq.fa ${file}-sort.bam | bcftools view -bvcg - > ${file}.bcf
#bcftools view ${file}.bcf > ${file}.vcf 

 
