#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=76:00:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=12

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l vmem=24gb

##PBS -q bigiron

#PBS -N beeewaretheHordeeum
#PBS -t 1-3

set -eux
umask 007


# launching compare-to-gff.prl on a single node so it can run till the cows come home
echo "Launching Blastn "

cd /lustre/groups/p2ep/rnaseq/parents/5stepBlast/

file=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/p2ep/rnaseq/parents/trinity/file.txt)

#for file in $(find ./ -iname "*.fna");
#do 

echo "${file}"

module load ncbi-blast+
#makeblastdb -in /lustre/groups/bioservices/hockeypuck/old/db/hordeumVulgare-Uniqunigene.fna -dbtype nucl

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/bioservices/hockeypuck/old/db/hordeumVulgare-Uniqunigene.fna -query ../trinity/${file}.Trinity.fasta -out ./tmp/${file}-bigHorde.xml -num_threads 12 -evalue 1e-20 -outfmt 5

#perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-bigHorde.xml ../trinity/try${file}/${file}.genes.results ../trinity/${file}.Trinity.fasta

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/bioservices/hockeypuck/old/db/TrichomonasVaginalis.fasta -query ./tmp/${file}-bigHorde.xml.e10noMatch.fasta -out ./tmp/${file}-trich.xml -num_threads 12 -evalue 1e-20 -outfmt 5

#perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-trich.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-bigHorde.xml.e10noMatch.fasta

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/p2ep/related/5step/Hordeum_vulgare.mRNA.PLN.fasta -query ./tmp/${file}-bigHorde.xml.e10noMatch.fasta -out ./tmp/${file}-hordeum2.xml -num_threads 12 -evalue 1e-10 -outfmt 5

#perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-hordeum2.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-bigHorde.xml.e10noMatch.fasta

#/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/p2ep/related/5step/triticum-unigene.fasta -query ./tmp/${file}-hordeum2.xml.e10noMatch.fasta -out ./tmp/${file}-triticum.xml -num_threads 12 -evalue 1e-10 -outfmt 5
 
#perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-triticum.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-hordeum2.xml.e10noMatch.fasta

/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/p2ep/related/5step/Oryza_sativa_Japonica_Group.mRNA.PLN.fasta -query ./tmp/${file}-triticum.xml.e10noMatch.fasta -out ./tmp/${file}-oryza.xml -num_threads 12 -evalue 1e-10 -outfmt 5

perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-oryza.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-triticum.xml.e10noMatch.fasta

/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/p2ep/related/5step/Zea_mays.mRNA.PLN.fasta -query ./tmp/${file}-oryza.xml.e10noMatch.fasta -out ./tmp/${file}-zeamays.xml -num_threads 12 -evalue 1e-10 -outfmt 5

perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-zeamays.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-oryza.xml.e10noMatch.fasta

/lustre/sw/ncbi-blast+/2.2.25/bin/blastn -db /lustre/groups/p2ep/related/5step/Arabidopsis_thaliana.mRNA.PLN.fasta -query ./tmp/${file}-zeamays.xml.e10noMatch.fasta -out ./tmp/${file}-arabid.xml -num_threads 12 -evalue 1e-10 -outfmt 5

perl ~/scripts/perl/xmlParseBlastAndKeep.pl ./tmp/${file}-arabid.xml ../trinity/try${file}/${file}.genes.results ./tmp/${file}-zeamays.xml.e10noMatch.fasta

echo `pwd`

#done




