#!/bin/bash
#PBS -l walltime=336:00:00  # Maximum amount of real time for the job to run in HH:MM:SS
#PBS -l nodes=1:ppn=64     # Number of nodes and processors per node requested by job
#PBS -l mem=1024GB           # Maximum physical memory to use for jobi
#PBS -N bbblasr-oat         # User-defined name for job
#PBS -j oe                # Write STDERR to STDOUT output file
#PBS -q hammerhead        # Job queue to submit this script to

# User prologue.  Must be owned by user, not writable by group or other
##PBS -l prologue=/projects/oat_genome/scripts/prologue.sh
# User epilogue.  Must be owned by user, not writable by group or other
##PBS -l epilogue=/projects/oat_genome/scripts/epilogue.sh
##PBS -t 7

IFS=$'\n'          # Internal Field Separator is newline
set -eu            # e - Exit on error
                   # u - Undefined variable is an error
                   # x - enable trace (print commands before running)
umask 007          # user  - all permissions
                   # group - all permissions
                   # other - no permissions
shopt -s nullglob  # Treat empty glob (*) as nothing instead of empty string


cd /nobackup/oat_genome/align


SEQUENCES=/nobackup/oat_genome/racon/allReads
ALIGNMENT=/nobackup/oat_genome/racon/aligned
GENOME_FASTA=/projects/oat_genome/dovetail-round2/oat_08Oct2018_80uBt.fasta
POLISHED_CONTIGS=/nobackup/oat_genome/racon/oat_racon_polished_contigs
THREADS=64

module load hdf5/1.8.19-gcc53
module load blasr

###concatenate all sequence lists together
printf "Starting aligning of sequences\n"

start=$(date +%s)

blasr /nobackup/oat_genome/racon/allReads.fq /projects/oat_genome/dovetail-round2/oat_08Oct2018_80uBt.fasta \
 --nproc 64 \
 --minMatch 15 \
 --maxMatch 20 \
 --advanceExactMatches 10 \
 --fastSDP \
 --aggressiveIntervalCut \
 --fastMaxInterval \
 --bam blasr-allreads.bam

end=$(date +%s)
printf "Done, took $((${end}-${start})) seconds\n\n"
