#!/bin/bash

# Request a run time of 5 hours and 30 minutes
#PBS -l walltime=72:30:00

# Request 1 processor in 1 node
#PBS -l nodes=1:ppn=32

# Request 7600 megabytes memory per processor.  ( 48 usable CPUs)
#PBS -l mem=64gb
#PBS -j oe
#PBS -N pppicard-drosophila
#PBS -q steelhead
#PBS -t 1-19
umask 007
set -eux

file=$(sed -n -e "${PBS_ARRAYID}p" /nobackup/rogers_research/rob/mayotte/files.txt)
#dir=$(sed -n -e "${PBS_ARRAYID}p" /lustre/groups/lorainelab/data/illumina/sweet_potato/filtered/dir.txt)

#module load java8 
#module load samtools
#module load bwa
#module load bcftools
module load picard/2.9.2

cd /nobackup/rogers_research/rob/mayotte/picard

#Getting just the first part of the $file id to use elsewhere.
NAME2=${file%_*}
name=${NAME2%_*}


#java -jar /users/rreid2/sw/picard/picard.jar ValidateSamFile I=${file}-sorted.bam R=/scratch/rreid2/banana/db/acuminata/GCA_000313855.2_ASM31385v2_genomic.fna O=val-${file}.txt MODE=SUMMARY
#java -jar /users/rreid2/sw/picard/picard.jar ValidateSamFile I=${file}-sorted.bam O=val-${file}.txt MODE=VERBOSE
java -jar /apps/pkg/picard-2.9.2/rhel7_u2-x86_64/gnu/libs/picard.jar AddOrReplaceReadGroups I=/nobackup/rogers_research/DsantSequences/AmirData/05_RLR_${file}_L004.sorted.bam \
VALIDATION_STRINGENCY=LENIENT \
O=/nobackup/rogers_research/rob/mayotte/picard/readgrouped-${file}.bam \
RGID=${name} \
RGLB=lib1 \
RGPL=illumina \
RGPU=unit1 \
RGSM=20 \
SORT_ORDER=coordinate \
CREATE_INDEX=true\

 
#samtools view -c -F 4 ${file}-sorted.bam > aligned-${file}.txt
#samtools view -c -f 4 ${file}-sorted.bam >> aligned-${file}.txt

#java -jar /users/rreid2/sw/picard/picard.jar CollectAlignmentSummaryMetrics \
#	R=/scratch/rreid2/banana/db/acuminata/GCA_000313855.2_ASM31385v2_genomic.fna \
#	I=${file}-sorted.bam  \
#	O=picardoutput-${file}.txt


#java -jar /users/rreid2/sw/picard/picard.jar BamIndexStats \
#	INPUT=${file}-sorted.bam  \
#	OUTPUT=picard-bamindexstats-${file}.txt

#bwa index -p PAHANG.idx -a bwtsw /scratch/rreid2/banana/db/acuminata/GCA_000313855.2_ASM31385v2_genomic.fna

#bwa aln  -t 12 PAHANG.idx /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R1_paired.fq > ${file}-1.aln.sai
#bwa aln  -t 12 PAHANG.idx /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R2_paired.fq > ${file}-2.aln.sai
#bwa sampe PAHANG.idx ${file}-1.aln.sai ${file}-2.aln.sai -P \
# /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R1_paired.fq \
# /scratch/rreid2/banana/filter/trimmomatic/trimgalored/output_${file}-R2_paired.fq  >  ${file}.sam

#samtools view -bS ${file}.sam | samtools sort -T ${file} -o ${file}-sorted.bam
#samtools mpileup -ugf /scratch/rreid2/banana/db/acuminata/GCA_000313855.2_ASM31385v2_genomic.fna ${file}-sorted.bam | bcftools view -vcg - > ${file}.bcf
#bcftools view ${file}.bcf > ${file}.vcf


#java -jar ~/sw/trimmomatic/Trimmomatic-0.36/trimmomatic-0.36.jar PE -threads 24 ../../trimgalore/${file}_R1_001_val_1.fq ../../trimgalore/${file}_R2_001_val_1.fq output_${file}-R1_paired.fq output_${file}-R1_unpaired.fq output_${file}-R2_paired.fq output_${file}-R2_unpaired.fq ILLUMINACLIP:TruSeq2-PE.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:50








